lib/ap/plan9/tmpfile.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * This file is part of the UCB release of Plan 9. It is subject to the license
+ * terms in the LICENSE file found in the top-level directory of this
+ * distribution and at http://akaros.cs.berkeley.edu/files/Plan9License. No
+ * part of the UCB release of Plan 9, including this file, may be copied,
+ * modified, propagated, or distributed except according to the terms contained
+ * in the LICENSE file.
+ */
+
+#include "stdio_impl.h"
+#include <unistd.h>
+#include <string.h>
+#include "sys9.h"
+#include "lib.h"
+#include "dir.h"
+
+#define TMPBASE "/tmp/tmpfile_"
+#define XXX "XXXXXX"
+
+FILE *
+tmpfile(void)
+{
+	FILE *f;
+	static char name[]={TMPBASE XXX};
+	int n;
+
+	snprintf(&name[sizeof(TMPBASE)], sizeof(XXX)-1, "%x", rand());
```

Comment:
Why not just,

    static char name[32];

    snprintf(name, sizeof(name), "/tmp/tmpfile_%x", rand());

?